app/model/commands/UpdateMetadataCommand.scala

```diff
@@ -0,0 +1,65 @@
+package model.commands
+
+import java.util.Date
+
+import com.gu.atom.data.PreviewDataStore
+import com.gu.atom.publish.PreviewAtomPublisher
+import com.gu.contentatom.thrift.{EventType, ContentAtomEvent}
+import com.gu.contentatom.thrift.atom.media.Platform
+import model.UpdatedMetadata
+import model.commands.CommandExceptions._
+import util.{YouTubeConfig, YouTubeVideoUpdateApi}
+import util.atom.MediaAtomImplicits
+
+import scala.util.{Failure, Success}
+
+case class UpdateMetadataCommand(atomId: String,
+                                 metadata: UpdatedMetadata)
+                                (implicit previewDataStore: PreviewDataStore,
+                                 previewPublisher: PreviewAtomPublisher,
+                                 val youtubeConfig: YouTubeConfig)
+    extends Command
+    with MediaAtomImplicits {
+
+  type T = Unit
+
+  def process(): Unit = {
+    previewDataStore.getAtom(atomId) match {
+      case Some(atom) =>
+        val mediaAtom = atom.tdata
+        val assets = mediaAtom.assets
+        val activeAsset = mediaAtom.activeVersion.flatMap(activeVersion => assets.find(_.version == activeVersion))
+
+        activeAsset match {
+          case Some(asset) =>
+
+            if(asset.platform == Platform.Youtube)
+              YouTubeVideoUpdateApi(youtubeConfig).updateMetadata(asset.id, metadata)
+
+            val newMetadata = mediaAtom.metadata.map(_.copy(
+              tags = metadata.tags,
+              categoryId = metadata.categoryId,
+              license = metadata.license))
```

Comment:
You're right, I was confused - as far as I know the non-youtube assets do not have metadata. I'll correct this.